### PR TITLE
feat: add Operator role

### LINKED ISSUE #327: [Feature Request]: Fine-grained per-user permissions within an organization
## Context

This follows up on #261 (Organizations & team-based access control), shipped in v1.2. The current implementation supports adding users to an organization, but as noted by @David-Crty there are no database server restrictions for users within the same org yet.

## Problem

In practice, not all members of an organization should have the same capabilities. Right now it's all-or-nothing inside an org. We need the ability to allow some users to work with backups (trigger, restore, browse) without giving them access to change server connection settings.

## Proposed permission model

Within an org, it should be possible to grant users scoped capabilities, for example:

- `viewer` — browse backups and snapshots only, no triggers
- `operator` — can trigger backups, restore, browse — but cannot edit server config
- `admin` — full access including server connection settings

Additionally, per-server or per-database scoping should be possible: a user in an org should optionally be limited to specific servers or even specific databases on a shared server.

## Use case

We have developers who manage application deployments but leave server/infrastructure management to us (IT). We want them to be able to:
- Trigger backups before a deployment
- Restore a specific database if something goes wrong
- Browse recent snapshots

But they should NOT be able to view or modify the server connection credentials or configuration. This needs to work both on dedicated servers and on shared servers where multiple clients' databases coexist.

## Notes for @David-Crty

This is essentially the fine-grained access layer you described in #261 (PR 2 in your proposed build order). Opening as a standalone issue to track separately. Happy to help spec out edge cases or test implementations.

### PR DESCRIPTION
**Pull request by David-Crty**
Adds the scoped-down **Operator** role from #327. (The Protected-server flag that was originally in this PR has been dropped to keep the change focused — it can be revisited separately.)

## What

A new `Operator` role sits between `Viewer` and `Member`. An Operator is "a Viewer that can also operate":

- ✅ Trigger backups
- ✅ Restore a database (incl. scheduled restores)
- ✅ Download snapshots
- ✅ Browse servers, snapshots & backups (same visibility as Viewer)
- ❌ Create/edit/delete server configuration & credentials
- ❌ Manage volumes, backup schedules
- ❌ Delete snapshots, manage users

Backed by a new `User::canOperate()` helper (Operator/Member/Admin/super-admin), leaving the existing `canPerformActions()` (Member/Admin) as the gate for configuration changes.

## Notes
- Per-user / per-server / per-database scoping is intentionally **not** included — Organizations already provide that boundary (as discussed in the issue).
- Added `Operator` to the user role picker, OAuth role mapping + priority order, and fr/es/el translations.
- Updated `docs/user-guide/permissions.md` with the new role and per-resource matrix.
- Refactored `DatabaseServer::buildExtraConfig()` to remove repeated set/unset/forget logic via a `pullExtraConfigKey()` helper.

## Tests
New `tests/Feature/Authorization/OperatorRoleTest.php` covers operator capabilities, viewer regression, and scheduled-restore management; plus an OIDC operator role-mapping test. Full suite green (1178 passed), PHPStan + Pint clean.

Closes #327

## Summary by CodeRabbit

* **New Features**
  * Added an **Operator** role across the app, including OAuth/OIDC mapping and role option/creation support.
  * Operators can run backups/restores, download/restore snapshots, and manage scheduled restores, but cannot edit server configuration.
* **Bug Fixes**
  * Updated authorization to consistently enforce Operator permissions for backup/restore/snapshot actions.
* **Documentation**
  * Refreshed permissions documentation and added localized **Operator** labels/descriptions.
* **Refactor & Tests**
  * Improved handling of type-specific server settings in `extra_config` and added automated tests for Operator permissions and config folding behavior.

### CODE REVIEW

File: app/Policies/RestorePolicy.php [hunk 1]
```diff
@@ -28,12 +28,12 @@ public function view(User $user, Restore|ScheduledRestore $restore): bool
...
      * Determine whether the user can start a new restore or schedule one.
-     * Demo users can create both. Final authorization on the target server
-     * is still checked via DatabaseServerPolicy@restore.
+     * Operators (and above) plus demo users can create. Final authorization on
+     * the target server is still checked via DatabaseServerPolicy@restore.
      */
     public function create(User $user): bool
     {
-        return $user->isDemo() || $user->canPerformActions();
+        return $user->isDemo() || $user->canOperate();
```
[the remark this answers is not in the document]
Comment (David-Crty) [reply app/Policies/RestorePolicy.php:36]: If a user can trigger restore, and it should be able to manage ScheduledRestore it's as simple as that
